### PR TITLE
 Convert ProtoAdapter.java to Kotlin

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+        if (value != null && value.isNotEmpty()) {
+          super.encodeWithTag(writer, tag, value)
+        }
+      }
+
+      override fun encodedSize(value: List<E>): Int {
+        var size = 0
+        var i = 0
+        val count = value.size
+        while (i < count) {
```
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: `for (i in 0 until value.size)`

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,583 @@
...
+
+      override fun encodedSizeWithTag(tag: Int, value: List<E>?): Int {
+        return if (value == null || value.isEmpty()) 0 else super.encodedSizeWithTag(tag, value)
+      }
+
+      @Throws(IOException::class)
+      override fun encode(writer: ProtoWriter, value: List<E>) {
+        var i = 0
+        val count = value.size
+        while (i < count) {
```
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: For

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,583 @@
...
+      override fun encodedSize(value: List<E>): Int {
+        throw UnsupportedOperationException("Repeated values can only be sized with a tag.")
+      }
+
+      override fun encodedSizeWithTag(tag: Int, value: List<E>?): Int {
+        if (value == null) return 0
+        var size = 0
+        var i = 0
+        val count = value.size
+        while (i < count) {
```
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: For

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,583 @@
...
+      override fun encode(writer: ProtoWriter, value: List<E>) {
+        throw UnsupportedOperationException("Repeated values can only be encoded with a tag.")
+      }
+
+      @Throws(IOException::class)
+      override fun encodeWithTag(writer: ProtoWriter, tag: Int, value: List<E>?) {
+        if (value == null) return
+        var i = 0
+        val count = value.size
+        while (i < count) {
```
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: For

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,583 @@
...
+      override fun encodedSize(value: Boolean): Int = FIXED_BOOL_SIZE
+
+      @Throws(IOException::class)
+      override fun encode(writer: ProtoWriter, value: Boolean) {
+        writer.writeVarint32(if (value) 1 else 0)
+      }
+
+      @Throws(IOException::class)
+      override fun decode(reader: ProtoReader): Boolean {
+        val value = reader.readVarint32()
```
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: `when (val value) {`

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,583 @@
...
+      }
+    }
+
+    @Throws(IOException::class)
+    override fun decode(reader: ProtoReader): Map<K, V> {
+      var key: K? = null
+      var value: V? = null
+
+      val token = reader.beginMessage()
+      var tag: Int
```
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Move into loop

Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Fixed

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt (hunk 5 above)
Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Fixed

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt (hunk 4 above)
Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Fixed

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt (hunk 3 above)
Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Fixed

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt (hunk 2 above)
Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Fixed

File: wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt (hunk 1 above)
Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt]: Fixed

**Pull request merged**